lib/streamlit/error_util.py

```diff
@@ -50,7 +51,21 @@ def handle_uncaught_app_exception(e: BaseException) -> None:
         # Use LOGGER.error, rather than LOGGER.debug, since we don't
         # show debug logs by default.
         LOGGER.error("Uncaught app exception", exc_info=e)
-        st.error(_GENERIC_UNCAUGHT_EXCEPTION_TEXT)
+        stacktrace_list = traceback.format_exc().splitlines()
+        # remove the exception message so we can keep information private
+        filtered_stacktrace_list = []
+        while stacktrace_list:
+            entry = stacktrace_list.pop()
+            if "Traceback (most recent call last):" in entry:
+                break
+            filtered_stacktrace_list.insert(0, entry)
+        # filtered_stacktrace_list = [entry for entry in stacktrace_list if not _is_in_streamlit_package(entry)]
+        filtered_stacktrace_list.pop()
+
+        new_exc = UncaughtAppException(Exception(filtered_stacktrace_list, exc_type))
```

Comment:
I think it'd be cleaner to just wrap the original exception in an `UncaughtAppException` and then handle the special casing in the st.exception code (so the code doing the stacktrace parsing above can get moved into `_get_stack_trace_str_list `). I also think that you might just be able to call `type(exc)` to get the exception type info in the `marshall` function so that you don't have to pass around `exc_type`